icechunk/src/session.rs

```diff
@@ -240,6 +242,26 @@ impl Session {
             .map(|node| self.change_set.update_array(node.id, metadata))
     }
 
+    pub async fn array_has_modified_chunks(
+        &self,
+        path: Path,
+        id: NodeId,
+    ) -> SessionResult<bool> {
+        let array = self.get_array(&path).await?;
```

Comment:
I think the caller always has executed get_array already? Maybe we should include in the arguments to this function the `NodeData`? I think with that you can also make this function return `bool`